### PR TITLE
Fix: Restrict the update profile API Endpoint from updating username.

### LINKED ISSUE #864: Restrict the update profile API Endpoint from updating username
Currently, django-rest-auth provides the functionality to update the username with the update profile API Endpoint. We want to restrict this since we don't want a user to update his username once he has created his account. 

For more details about the API endpoint, look at http://django-rest-auth.readthedocs.io/en/latest/api_endpoints.html#basic

### PR DESCRIPTION
**Pull request by isht3**
Fixes #864 

* Override the default django-rest-auth serializer by adding
  REST_AUTH_SERIALIZERS to settings and adding the location to
  the new serializer.
* Make a custom serializer in accounts app, and add the username field
  as a read_only_fields.

The test for the fix is being written. :)

### CODE REVIEW

**Comment by deshraj** [comment]
@isht3 looks good. I will merge this whenever you push the test. :)

**Comment by isht3** [comment]
@xamfy I have some problem with running tests in my system. Would you like to do it? I'll guide you through it.

**Comment by xamfy** [comment]
@isht3 Sure.

**Comment by isht3** [comment]
@xamfy Let me know on gitter, when you're ready. :)

**Comment by isht3** [comment]
@deshraj Since I'm not able to do the testing part, I'm guiding @xamfy through it, so I think this PR can be reviewed and merged. :)

**Comment by RishabhJain2018** [comment]
@isht3 The code looks good to me! Let's merge this when @xamfy is ready with test cases.

File: settings/common.py [hunk 1]
```diff
@@ -273,3 +273,8 @@
     },
     'SUBMISSION_QUEUE': 'submission_task_queue',
 }
+
+# To make usermame field read-only, customer serializer is defined.
```
Comment (RishabhJain2018) [inline settings/common.py]: `customer` -> `customized`

**Comment by xamfy** [comment]
@RishabhJain2018 Done with test cases. Should I send a separate PR for tests only?

**Comment by isht3** [comment]
@xamfy Sure, go ahead!

**Comment by RishabhJain2018** [comment]
LGTm :) Thanks for the fix!
@deshraj Please merge #1281 first & then #1283 .

**Comment by xamfy** [comment]
@RishabhJain2018 #1281 has been closed

**Comment by RishabhJain2018** [comment]
Ohh..Thanks for the update.

File: apps/accounts/serializers.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+# Get the UserModel
+UserModel = get_user_model()
+
+
+class UserDetailsSerializer(serializers.ModelSerializer):
+    """
+    Make username as a read_only field.
+    """
+    class Meta:
+        model = UserModel
```
Comment (deshraj) [inline apps/accounts/serializers.py]: We could directly do `model=get_user_model()` instead of storing it in another variable.

Comment (isht3) [reply apps/accounts/serializers.py]: @deshraj Done!

**Comment by xamfy** [comment]
Any progress with this?

**Comment by RishabhJain2018** [comment]
@xamfy Don't Worry. we will merge this :)

**Comment by deshraj** [comment]
Great job @isht3 :)

**Pull request merged**